### PR TITLE
# issue 33 - Item crud

### PR DESCRIPTION
**Pull request by PragatiDoshi-BTC**

### CODE REVIEW

File: app/controllers/items_controller.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      flash.clear
+      flash[:error] = "Item not updated because: #{@item.errors.full_messages.join(',')}"
+    end
+  end
+
+  def destroy
+    if @item.destroy
+      flash[:success] = "Item deleted successfully"
+    else
+      flash[:error] = "Item not deleted because: #{@item.errors.full_messages.join(',')}"
```
Comment (amitpatelx) [inline app/controllers/items_controller.rb]: message should be `Unable to delete the item`

File: app/controllers/items_controller.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    end
+  end
+
+  def update
+    if @item.update(item_params)
+      flash.clear
+      flash[:success] = "Item updated successfully"
+    else
+      flash.clear
+      flash[:error] = "Item not updated because: #{@item.errors.full_messages.join(',')}"
```
Comment (amitpatelx) [inline app/controllers/items_controller.rb]: message should be `Unable to update the item`

File: app/models/item.rb [hunk 3]
```diff
@@ -1,3 +1,8 @@
 class Item < ActiveRecord::Base
   has_many :taxes
+
+  validates :name, presence: true
+  validates :price, presence: true
+  validates :price, format: { :with => /\A\d+(?:\.\d{0,2})?\z/ }, numericality: {greater_than: 0, only_integer: false}
```
Comment (amitpatelx) [inline app/models/item.rb]: The should be space after `{` and before `}`

File: app/views/items/index.html.haml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+              %tr
+                %th Name
+                %th Description
+                %th Price
+                %th Actions
+            - @items.each do |item|
+              %tr
+                %td= item.name
+                %td= item.description
+                %td= number_with_precision(item.price, :precision => 2)
```
Comment (amitpatelx) [inline app/views/items/index.html.haml]: Require Ruby 2 hash notation

File: app/views/items/index.html.haml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+                %td
+                  = link_to("Edit",edit_item_path(item),"data-toggle" => :modal, remote: true)
```
Comment (amitpatelx) [inline app/views/items/index.html.haml]: Can you please check if `data: { toggle: :modal }` works?

File: app/views/items/update.js.haml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+- if flash[:error].present?
```
Comment (amitpatelx) [inline app/views/items/update.js.haml]: This script looks similar to `create.js.haml`. Try to reuse

**Comment by ParthivPatel-BTC** [comment]
@PragatiDoshi-BTC - Flash messages not working properly of Item Add/Edit form.

selection_023

If I Click on "Cancel" button then clicking on "Users" link form the left sidebar it shows **error messages on users index page as well**. Please resolve this.

selection_025

selection_027

**Pull request merged**